www/src/pages/editor/NewMineralSiteModal.tsx

```diff
@@ -0,0 +1,409 @@
+          source: currentUserUrl,
+          normalizedURI: values.stateOrProvince,
+          confidence: 1.0,
+        }),
+      ]
+      : [];
+
+    const commodity1 = commodity?.id;
+    const referenceDocument = new Document({
+      uri: values.refDoc,
+    });
+
+    const reference = new Reference({
+      document: referenceDocument,
+      comment: values.refComment,
+    });
+
+    const mineralInventory = new MineralInventory({
+      category: ["Inferred", "Indicated", "Measured"].map(
+        (name) =>
+          new CandidateEntity({
+            source: currentUserUrl,
+            confidence: 1.0,
+            observedName: name,
+            normalizedURI: `https://minmod.isi.edu/resource/${name}`,
+          })
+      ),
+      commodity: new CandidateEntity({
+        source: currentUserUrl,
+        confidence: 1.0,
+        observedName: commodity?.name,
+        normalizedURI: commodity?.uri,
+      }),
+      grade: values.grade !== undefined && values.grade !== null
+        ? new Measure({
+          value: values.grade,
+          unit: new CandidateEntity({
+            source: currentUserUrl,
+            confidence: 1,
+            observedName: unitStore.getByURI(values.gradeUnit!)!.name,
+            normalizedURI: values.gradeUnit,
+          }),
+        })
+        : undefined,
+      ore: values.tonnage !== undefined && values.tonnage !== null
+        ? new Measure({
+          value: values.tonnage,
+          unit: new CandidateEntity({
+            source: currentUserUrl,
+            confidence: 1,
+            observedName: unitStore.getByURI(values.tonnageUnit!)!.name,
+            normalizedURI: values.tonnageUnit,
+          }),
+        })
+        : undefined,
+      reference: reference,
+    });
+
+
+    const sourceType = values.sourceType;
+    const refDocUrl = values.refDoc;
+    let combinedSourceId = "";
+
+    if (sourceType === "unpublished") {
+      combinedSourceId = `unpublished::${currentUserUrl}`;
+    } else if (sourceType !== "unpublished") {
+      combinedSourceId = `${sourceType}::${refDocUrl}`;
+    }
+
+
+    const draft = new DraftCreateMineralSite({
+      id: "",
+      draftID: `draft-${Date.now()}`,
+      recordId: `record-${uuidv4()}`,
+      sourceId: combinedSourceId,
+      dedupSiteURI: "",
+      createdBy: [currentUserUrl],
+      name: values.name,
+      locationInfo: new LocationInfo({
+        country: countries,
+        stateOrProvince: statesOrProvinces,
+        location: location,
+      }),
+      depositTypeCandidate: [
+        new CandidateEntity({
+          source: currentUserUrl,
+          confidence: values.depositTypeConfidence,
+          observedName: depositTypeStore.getByURI(values.depositType)!.name,
+          normalizedURI: values.depositType,
+        }),
+      ],
+      reference: [reference],
+      sameAs: [],
+      gradeTonnage: {
+        [commodity1 as string]: new GradeTonnage({
+          commodity: commodity1 ?? " ",
```

Comment:
problematic code, if you need type cast; it usually means you are doing something wrong. in this case, it is because commodity can be undefined.